tests/smoke-tests/--typescript.test.ts

```diff
@@ -1,58 +1,79 @@
-import fs from 'node:fs/promises';
+import { execa } from 'execa';
 import path from 'node:path';
 import { afterAll, beforeAll, describe, expect, it } from 'vitest';
 
 import {
+  AddonHelper,
   assertGeneratedCorrectly,
-  createAddon,
-  createTmp,
   dirContents,
-  install,
-  runScript,
   SUPPORTED_PACKAGE_MANAGERS,
 } from '../helpers.js';
 
 for (let packageManager of SUPPORTED_PACKAGE_MANAGERS) {
   describe(`--typescript with ${packageManager}`, () => {
-    let cwd = '';
-    let tmpDir = '';
     let distDir = '';
     let declarationsDir = '';
+    let helper = new AddonHelper({
+      packageManager,
+      args: ['--typescript'],
+      scenario: 'typescript',
+    });
 
     beforeAll(async () => {
-      tmpDir = await createTmp();
-
-      let { name } = await createAddon({
-        args: ['--typescript', `--${packageManager}=true`, '--skip-npm'],
-        options: { cwd: tmpDir },
-      });
+      await helper.setup();
+      await helper.installDeps();
 
-      cwd = path.join(tmpDir, name);
-      distDir = path.join(cwd, name, 'dist');
-      declarationsDir = path.join(cwd, name, 'declarations');
-
-      await install({ cwd, packageManager, skipPrepare: true });
+      distDir = path.join(helper.addonFolder, 'dist');
+      declarationsDir = path.join(helper.addonFolder, 'declarations');
     });
 
     afterAll(async () => {
-      await fs.rm(tmpDir, { recursive: true, force: true });
+      await helper.clean();
     });
 
     it('was generated correctly', async () => {
-      await runScript({ cwd, script: 'build', packageManager: 'pnpm' });
+      await helper.build();
 
-      assertGeneratedCorrectly({ projectRoot: cwd });
+      assertGeneratedCorrectly({ projectRoot: helper.projectRoot });
     });
 
-    it('builds the addon', async () => {
-      let { exitCode } = await runScript({ cwd, script: 'build', packageManager: 'pnpm' });
+    // Tests are additive, so when running them in order, we want to check linting
+    // before we add files from fixtures
+    it('lints all pass', async () => {
+      let { exitCode } = await helper.run('lint');
 
       expect(exitCode).toEqual(0);
+    });
+
+    it('build and test', async () => {
+      // Copy over fixtures
+      await helper.fixtures.use('./my-addon/src/components');
+      await helper.fixtures.use('./test-app/tests');
```

Comment:
I think so.
May make more sense to be like 'useAll' or something to that effect.

I didn't use everything because i wanted to
- avoid other files (such as top-level, helpers, etc)
- be slightly more resistant to potential other changes in the fixtures